rust/src/types/node.rs

```diff
@@ -0,0 +1,180 @@
+}
+
+impl<const N: usize, C> Default for SparseLeafNode<N, C>
+where
+    C: Commitment + Default,
+{
+    fn default() -> Self {
+        let mut values = [ValueWithIndex::default(); N];
+        values.iter_mut().enumerate().for_each(|(i, v)| {
+            v.index = i as u8;
+        });
+
+        SparseLeafNode {
+            commitment: C::default(),
+            stem: [0; 31],
+            values,
+        }
+    }
+}
+
+/// A leaf node with 256 children in a (file based) Verkle trie.
+// NOTE: This type should NOT implement [`Clone`] because there should never be two instances
+// corresponding to the same logical node.
+// NOTE: Changing the layout of this struct will break backwards compatibility of the
+// serialization format.
+#[derive(Debug, PartialEq, Eq, FromBytes, IntoBytes, Immutable)]
+#[cfg_attr(test, derive(Clone))] // For testing purposes only
+#[repr(C)]
+pub struct FullLeafNode<C: Commitment> {
+    pub commitment: C,
+    pub stem: [u8; 31],
+    pub values: [Value; 256],
+}
+
+impl<C> Default for FullLeafNode<C>
+where
+    C: Commitment + Default,
+{
+    fn default() -> Self {
+        FullLeafNode {
+            commitment: C::default(),
+            stem: [0; 31],
+            values: [Value::default(); 256],
+        }
+    }
+}
+
+/// An inner node in a (file based) Verkle trie.
+// NOTE: This type should NOT implement [`Clone`] because there is never be two instances
+// corresponding to the same logical node.
+// NOTE: Changing the layout of this struct will break backwards compatibility of the
+// serialization format.
+#[derive(Debug, PartialEq, Eq, FromBytes, IntoBytes, Immutable, Unaligned)]
+#[cfg_attr(test, derive(Clone))] // For testing purposes only
+#[repr(C)]
+pub struct InnerNode<C: Commitment> {
+    pub commitment: C,
+    pub values: [NodeId; 256],
+}
+
+impl<C> Default for InnerNode<C>
+where
+    C: Commitment + Default,
+{
+    fn default() -> Self {
+        InnerNode {
+            commitment: C::default(),
+            values: [NodeId::from_idx_and_node_type(0, NodeType::Empty); 256],
+        }
+    }
+}
+
+/// A node in a (file based) Verkle trie.
+// NOTE: This type should NOT implement [`Clone`] because there should never be two instances
+// corresponding to the same logical node.
+#[derive(Debug, PartialEq, Eq)]
+#[cfg_attr(test, derive(Clone))] // For testing purposes only
+pub enum Node<C: Commitment> {
+    Empty,
+    Inner(Box<InnerNode<C>>),
+    Leaf2(Box<SparseLeafNode<2, C>>),
+    Leaf256(Box<FullLeafNode<C>>),
+}
+
+/// A node type of a node in a (file based) Verkle trie.
+/// This type is primarily used for conversion between [`Node`] and indexes in the file storage.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum NodeType {
+    Empty,
+    Inner,
+    Leaf2,
+    Leaf256,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn sparse_leaf_node_default_returns_leaf_node_with_all_values_set_to_default_and_unique_indices()
+     {
```

Comment:
Uhm, why was this not caught by CI? Line too long?

Reply:
What should be caught? The line is auto formatted.